src/Functions/FunctionsComparison.h

```diff
@@ -1232,11 +1235,20 @@ class FunctionComparison : public IFunction
             }
             else
             {
-                // compare
+                /// Check does another data type is comparable to Decimal, includes Int and Float.
                 if (!allowDecimalComparison(left_type, right_type) && !date_and_datetime)
                     throw Exception(
                         "No operation " + getName() + " between " + left_type->getName() + " and " + right_type->getName(),
                         ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+                /// When Decimal comparing to Float32/64, we convert both of them into Float64.
+                /// Other systems like MySQL and Spark also do as this.
```

Comment:
Will it work correctly for `Decimal128` and `Decimal256`?

Reply:
Yes, it also work well for `Decimal128` and `Decimal256`.